geomesa-nifi-processors/src/main/scala/org/locationtech/geomesa/nifi/GeoMesaIngestProcessor.scala

```diff
@@ -59,24 +60,41 @@ class GeoMesaIngestProcessor extends AbstractProcessor {
   @volatile
   private var converter: convert.SimpleFeatureConverter[_] = null
 
-
   @OnScheduled
   def initialize(context: ProcessContext): Unit = {
-    dataStore = getDataStore(context)
-    val sft = getSft(context)
-    dataStore.createSchema(sft)
 
-    converter = getConverter(sft, context)
-    featureWriter = createFeatureWriter(sft, context)
-    getLogger.info(s"Initialized GeoMesaIngestProcessor datastore, fw, converter for type ${sft.getTypeName}")
+    try {    
+      val zookeepers = context.getProperty(Zookeepers).getValue
+      val nc_host = "nc " + zookeepers.replace(':', ' ')
+      val ret = ("echo ruok" #| nc_host)!!
+
+      dataStore = getDataStore(context)
+      val sft = getSft(context)
+      dataStore.createSchema(sft)
+      converter = getConverter(sft, context)
+      featureWriter = createFeatureWriter(sft, context)
+      getLogger.info(s"@hangfix Initialized GeoMesaIngestProcessor datastore, fw, converter for type ${sft.getTypeName}")
+    } catch {
+      case e: Exception =>
+        featureWriter = null
+        getLogger.info(s"@hangfix Error: the context is not availalbe, zookeepers may not be running, dataStore equals ${dataStore}")
+    }
   }
 
   @OnStopped
   def cleanup(): Unit = {
-    IOUtils.closeQuietly(featureWriter)
-    featureWriter = null
-    dataStore = null
-    getLogger.info("Shut down geomesa processor")
+    try {
+	featureWriter = null
```

Comment:
this is the wrong order...the feature writer needs to be closed first before the reference is nulled out